File: WEEK 1/Python/Dairy Farm Profit Calculation .py

LITERS_PER_CARTON = 3.78
COST_PER_LITER = 0.38
PROFIT_PER_CARTON = 0.27

def calculate_profit(total_liters):

    cartons_needed = total_liters / LITERS_PER_CARTON
    cost = total_liters * COST_PER_LITER
    profit = cartons_needed * PROFIT_PER_CARTON
    return cartons_needed, cost, profit


while True:
    try:

        total_liters = float(input("Enter the total amount of milk produced in the morning (in liters): "))

        if total_liters <= 0:
            print("Please enter a positive value for the total milk produced.")
        else:

            cartons_needed, cost, profit = calculate_profit(total_liters)
            print(f"Number of milk cartons needed: {cartons_needed:.2f}")
            print(f"Cost of producing the milk: ${cost:.2f}")
            print(f"Profit for producing the milk: ${profit:.2f}")
            break

    except ValueError:
        print("Invalid input. Please enter a valid number for the total milk produced.")

File: WEEK 1/JavaScript/Dairy Farm Profit Calculation.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function getUserInput(prompt) {
  return new Promise((resolve) => {
    rl.question(prompt, (input) => {
      resolve(parseFloat(input));
    });
  });
}

async function main() {
  try {
    const totalMilkProduced = await getUserInput('Enter the total amount of milk produced in liters in the morning: ');

    if (totalMilkProduced < 0) {
      console.log('Please enter a non-negative value for milk production.');
      return;
    }

    const milkCartonCapacity = 3.78; // liters
    const costPerLiter = 3.78; // ZAR
    const profitPerCarton = 2.70; // ZAR

    const cartonsNeeded = Math.ceil(totalMilkProduced / milkCartonCapacity);
    const totalCost = totalMilkProduced * costPerLiter;
    const totalProfit = cartonsNeeded * profitPerCarton;

    console.log(`Number of milk cartons needed: ${cartonsNeeded}`);
    console.log(`Cost of producing the milk: ZAR ${totalCost.toFixed(2)}`);
    console.log(`Profit for producing the milk: ZAR ${totalProfit.toFixed(2)}`);
  } catch (error) {
    console.error(error);
  } finally {
    rl.close();
  }
}

main();
